### PR TITLE
Minor pathfinder-related changes (preparation for mobile refactor)

### PR DESCRIPTION
**Pull request by reaperrr**
This is a bunch of presumably "safe", easy to review changes from #7565.

### CODE REVIEW

**Comment by Mailaender** [comment]
Looks good. :+1: Can't notice any changes in-game. :white_check_mark:

File: OpenRA.Game/Actor.cs [hunk 1]
```diff
@@ -172,6 +172,11 @@ public void QueueActivity(Activity nextActivity)
 				currentActivity.Queue(nextActivity);
 		}
 
+		public bool IsAlliedWith(Actor actor)
```
Comment (pchote) [inline OpenRA.Game/Actor.cs]: I'm not convinced about this.  Alliances are defined in terms of players, not actors, and this just adds a duplicate code path.

**Comment by reaperrr** [comment]
Updated. Travis error is not my fault.
Removed `IsAlliedWith` and additionally removed an `as World` and `as Actor` from PathFinder which I had previously overlooked.

File: OpenRA.Mods.Common/Pathfinder/PathGraph.cs [hunk 2]
```diff
@@ -127,17 +127,17 @@ int GetCostToNode(CPos destNode, CVec direction)
...
 				out movementCost,
-				IgnoredActor as Actor,
+				IgnoredActor,
 				checkConditions) && !(CustomBlock != null && CustomBlock(destNode)))
 			{
 				return CalculateCellCost(destNode, direction, movementCost);
 			}
 
-			return InvalidNode;
+			return Constants.InvalidNode;
```
Comment (penev92) [inline OpenRA.Mods.Common/Pathfinder/PathGraph.cs]: If you are replacing the `InvalidNode` from this class with `Constants.InvalidNode`, why are you not removing the old one?

Comment (reaperrr) [inline OpenRA.Mods.Common/Pathfinder/PathGraph.cs]: I think the original refactor PR forgot about that too (or I just overlooked it). Anyway, fixed.

**Comment by obrakmann** [comment]
:+1:

**Comment by reaperrr** [comment]
Huh? Does github auto-rebase now? Because this definitely had some changes overlapping with #8622.

**Pull request merged**